corehq/apps/userreports/sql/adapter.py

```diff
@@ -145,16 +148,30 @@ def handle_exception(self, doc, exception):
 def get_indicator_table(indicator_config, custom_metadata=None):
     sql_columns = [column_to_sql(col) for col in indicator_config.get_columns()]
     table_name = get_table_name(indicator_config.domain, indicator_config.table_id)
+    extra_indices = [
+        Index(
+            _custom_index_name(table_name, index.column_ids),
+            *index.column_ids
+        )
+        for index in indicator_config.sql_column_indexes
+    ]
+    columns_and_indices = sql_columns + extra_indices
```

Comment:
not blocking but you should validate that the columns specified in the index are actual columns in the table.